SAML Auth Keycloak - I can enable keycloak but if I log out and login again I'm getting an error

Enabling Keycloak is working in this same setup using Ember UI.

I noticed that when enabling from Dashboard I click enable and I don't get the usual Pop Up we get in Ember UI to authenticate with a User to finish enabling the Auth. 

This also seems to create a principal which uid is invalid for the auth

`[ERROR] SAML: Failed getting user with error: can't find unique user for principal keycloak_user://testuser1@keycloak.com`

Steps.

- Follow the steps from the official Rancher docs  https://rancher.com/docs/rancher/v2.x/en/admin-settings/authentication/keycloak/
- Enable the Keycloak Auth
- Notice there isn't a pop up appearing but the Keycloak auth is enabled.
- Logout
- Try logging with a valid Keycloak user 

![Screen Shot 2021-01-21 at 3 49 30 PM](https://user-images.githubusercontent.com/27659/105422617-43f84980-5c01-11eb-96ac-59bd3574cfd6.png)


Reference: https://github.com/rancher/dashboard/issues/2115
Rancher version `v2.5-head (01/21/2021)` _23a29c306_

--- a/cypress/e2e/tests/pages/fleet/fleet-clusters.spec.ts
+++ b/cypress/e2e/tests/pages/fleet/fleet-clusters.spec.ts
@@ -505,8 +505,7 @@ describe('Fleet CLuster List - resources', { tags: ['@fleet', '@adminUser'] }, (
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
 
   it('should display fleet clusters list page', () => {

--- a/cypress/e2e/tests/pages/fleet/gitrepo.spec.ts
+++ b/cypress/e2e/tests/pages/fleet/gitrepo.spec.ts
@@ -417,8 +417,7 @@ describe('Git Repo', { testIsolation: 'off', tags: ['@fleet', '@adminUser'] }, (
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
   it('should display continuous delivery page git repo', () => {
     const gitRepoList = new FleetGitRepoListPagePo();

--- a/cypress/e2e/tests/pages/manager/cloud-credential.spec.ts
+++ b/cypress/e2e/tests/pages/manager/cloud-credential.spec.ts
@@ -328,8 +328,7 @@ describe('Cloud Credential', { tags: ['@manager', '@adminUser'] }, () => {
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   beforeEach(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
     HomePagePo.goTo(); // this is needed to ensure we have a valid authentication session
   });
 

--- a/cypress/e2e/tests/pages/manager/cluster-manager.spec.ts
+++ b/cypress/e2e/tests/pages/manager/cluster-manager.spec.ts
@@ -822,8 +822,7 @@ describe('Cluster Manager as standard user', { testIsolation: 'off', tags: ['@ma
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
   it('should display cluster manager page', () => {
     const clusterList = new ClusterManagerListPagePo();

--- a/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
+++ b/cypress/e2e/tests/pages/manager/jwt-authentication.spec.ts
@@ -191,8 +191,7 @@ describe('JWT Authentication (Standard User)', { testIsolation: 'off', tags: ['@
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
 
   it('should display JWT Authentication empty list page', () => {

--- a/cypress/e2e/tests/pages/manager/kontainer-drivers.spec.ts
+++ b/cypress/e2e/tests/pages/manager/kontainer-drivers.spec.ts
@@ -289,8 +289,7 @@ describe('Kontainer Drivers', { testIsolation: 'off', tags: ['@manager', '@admin
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
 
   it('should display kontainer drivers list page', () => {

--- a/cypress/e2e/tests/pages/manager/machines.spec.ts
+++ b/cypress/e2e/tests/pages/manager/machines.spec.ts
@@ -135,8 +135,7 @@ describe('Machines', { testIsolation: 'off', tags: ['@manager', '@adminUser'] },
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
 
   it('should display machines list page', () => {

--- a/cypress/e2e/tests/pages/manager/pod-security-admissions.spec.ts
+++ b/cypress/e2e/tests/pages/manager/pod-security-admissions.spec.ts
@@ -170,8 +170,7 @@ describe('Pod Security Admissions', { testIsolation: 'off', tags: ['@manager', '
 describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () => {
   before(() => {
     cy.login();
-    // Set theme to light
-    cy.setUserPreference({ theme: 'ui-light' });
+    cy.applyDefaultTestTheme();
   });
 
   it('should display Pod Security Admissions list page', () => {
@@ -185,7 +184,7 @@ describe('Visual Testing', { tags: ['@percy', '@manager', '@adminUser'] }, () =>
     podSecurityAdmissionsPage.list().resourceTable().sortableTable().noRowsShouldNotExist();
 
     // hide elements before taking percy snapshot
-    cy.hideElementBySelector('[data-testid="nav_header_showUserMenu"]', '[data-testid="type-count"]');
+    cy.hideElementBySelector('[data-testid="nav_header_showUserMenu"]', '[data-testid="type-count"]', 'td.col-live-date span.live-date');
 
     // takes percy snapshot.
     cy.percySnapshot('Pod Security Admissions list page');

--- a/cypress/globals.d.ts
+++ b/cypress/globals.d.ts
@@ -121,7 +121,7 @@ declare global {
          */
         wait?: number,
       }): Chainable;
-
+      applyDefaultTestTheme(): Chainable<any>;
       getRancherVersion(): Chainable<any>;
       getRancherResource(prefix: 'v3' | 'v1', resourceType: string, resourceId?: string, expectedStatusCode?: number): Chainable;
       setRancherResource(prefix: 'v3' | 'v1', resourceType: string, resourceId: string, body: any): Chainable;

--- a/cypress/support/commands/rancher-api-commands.ts
+++ b/cypress/support/commands/rancher-api-commands.ts
@@ -1402,3 +1402,40 @@ Cypress.Commands.add('getClusterIdByName', (clusterName: string) => {
     }
   });
 });
+
+/**
+ * Cypress custom command: applyDefaultTestTheme
+ *
+ * Sets the Rancher UI to use the modern branding theme and applies the light (white) theme for the current user.
+ *
+ * Steps performed:
+ * 1. Updates the Rancher "ui-brand" setting via the v3 API to "modern".
+ * 2. Sets the user's UI preference theme to "ui-light".
+ *
+ * Usage:
+ *    cy.applyDefaultTestTheme();
+ *
+ * Notes:
+ * - Requires that cy.setRancherResource and cy.setUserPreference commands are defined.
+ * - Uses the v3 Rancher API, which avoids conflicts with resourceVersion.
+ * - Ideal for automation tests where a consistent UI theme is needed.
+ */
+Cypress.Commands.add('applyDefaultTestTheme', () => {
+  // Step 1: Set Rancher branding
+  cy.setRancherResource('v3', 'settings', 'ui-brand', { value: 'modern' })
+    .then((response) => {
+      Cypress.log({
+        name:    'setRancherResource',
+        message: `Rancher UI brand set to: ${ response.value || 'MODERN' }`
+      });
+    });
+
+  // Step 2: Set user preference for light theme
+  cy.setUserPreference({ theme: 'ui-light' })
+    .then((response) => {
+      Cypress.log({
+        name:    'setUserPreference',
+        message: `User theme preference set to: ${ response.theme || 'ui-light' }`
+      });
+    });
+});

--- a/package.json
+++ b/package.json
@@ -155,7 +155,7 @@
     "@cypress/grep": "3.1.5",
     "@cypress/vue": "5.0.5",
     "@cypress/webpack-dev-server": "3.4.1",
-    "@percy/cli": "1.30.7",
+    "@percy/cli": "1.31.8",
     "@percy/cypress": "3.1.6",
     "@types/copy-webpack-plugin": "5.0.3",
     "@types/dompurify": "3.0.5",

